x-pack/plugins/fleet/server/routes/security.ts

```diff
@@ -81,16 +83,36 @@ export async function getAuthzFromRequest(req: KibanaRequest): Promise<FleetAuth
     );
     const fleetSetupAuth = getAuthorizationFromPrivileges(privileges.kibana, 'fleet-setup');
 
+    const writeEndpointIntegrationAuth = getAuthorizationFromPrivileges(
+      privileges.kibana,
+      `${INTEGRATIONS_PLUGIN_ID}-writeEndpointIntegration`
+    );
+
+    const readEndpointIntegrationAuth = getAuthorizationFromPrivileges(
+      privileges.kibana,
+      `${INTEGRATIONS_PLUGIN_ID}-readEndpointIntegration`
+    );
+
     return calculateAuthz({
       fleet: { all: fleetAllAuth, setup: fleetSetupAuth },
-      integrations: { all: intAllAuth, read: intReadAuth },
+      integrations: {
+        all: intAllAuth,
+        read: intReadAuth,
+        writeEndpointIntegration: writeEndpointIntegrationAuth,
```

Comment:
Added read/write endpoint integration privilege under `Integrations` category as this was the simplest change. If we were to add it under `Security` main category, the code needs more changes for `FleetAuthz` object to store security privileges in a new property.